core/src/main/java/com/linecorp/armeria/internal/common/util/ServiceNamingUtil.java

```diff
@@ -39,5 +39,17 @@ public static String fullTypeHttpServiceName(HttpService service) {
         }
     }
 
+    public static String trimTrailingDollarSigns(String serviceName) {
+        int lastIndex = serviceName.length() - 1;
+        if (serviceName.charAt(lastIndex) != '$') {
+            return serviceName;
+        }
+
+        do {
+            lastIndex--;
+        } while (serviceName.charAt(lastIndex) == '$');
```

Comment:
This will raise an IOOBE if a service name is `$` :laughing: Could you fix this and add a test case?

Reply:
I added the condition `lastIndex > 0` here to handle not only `$` but also multiple signs such as `$$$`.